### DISCUSSION TITLE
Does this integration work anymore?

### DISCUSSION THREAD

**Discussion by mikemccomas** [alandtse/alexa_media_player#2315] [General]
It worked for a few months but now, both of my Echo devices are no longer available. It says Failed to set up. Tried re-logging in to Amazon and it still said Failed to set up. Tried removing and re-installing. Now it just says 500 error, the server got into trouble whenever I try to log in and complete the re-captcha.

**Comment by nicedevil007** [comment]
Same on my site. I got this integration in trouble in late june 2024. Today I realized that most of my echo devices are there with double entries. So I removed the integration, restarted homeassistant and got the same as @mikemccomas.
It was redirecting to the page where we have to enter the captcha and after this it says Server error 500, The server got itself in trouble.

**Reply by nicedevil007** [reply]
maybe I fixed it now, but then the description is wrong?

@mikemccomas 
Just want to let you know that I'm using amazon.de because I live in germany ofc.
So the URL for amazon is by default amazon.com and I always changed it as the example in the brackets tell us to amazon.de.

This lead to the behaviour above!

I rebooted my HASS, added the integration again and left the amazon.com as the default value, now it works!

**Reply by mikemccomas** [reply]
I live in the US so I just left it as amazon.com. I tried completely uninstalling the integration (instead of just removing the devices), then rebooting HA but still no luck,

**Reply by rolox** [reply]
I do't think it's a chane in the Amazon API. I went back to HA 2024.7.4 and Alexa Media Player works again. At least after reconfiguring it.

**Comment by f3bandit** [comment]
same trying to set it up after upgrading my hardware to HA yellow.
I get a 404 error after setting up the 2FA code with the integration,
get the 404 on the open webpage button shows up after confirming
adding the 2FA code.

**Comment by Armand29** [comment]
It isn’t working for me either.  When I try to enable it it gives me an ‘unknown error”.  It was working a few weeks ago, but now I cannot enable it at all.

**Comment by solarjunkev** [comment]
Me either. Thanks to everyone for taking the time to attempt to fix this.
AMP is one of my most commonly used integrations
Questions from a layman
1/ Is this problem due to a failure of Alexa two factor authentication?
2/ Why did this suddenly become a problem? Have Amazon changed anything (their end)?
3/ Does it look like an easy fix (in a future HA update) or should we expect this to be down for weeks/months?
Thanks!

**Comment by Armand29** [comment]
Tried everything and all I get is “Unknown error”.   Please fix!!

**Comment by Armand29** [comment]
Started working after latest update!  Did a full re-install and it connected.

**Reply by bellyup64** [reply]
It stopped working for me after the last update. I'm going to rollback when I get home to test. 
20.24.8.0 got me with update badges... usually I wait and not jump on .0 release.

**Comment by mpoffo1231** [comment]
Same issues as reported by others.  Setup failed message when loading.  Reloading it use to "fix" this issue but it simply does not work now.  Try to delete and reinstall.  Encountered the same issues other reported with the 500 error.  I am not sure what changed other than the latest HA assistant update today.  I had updated the Alexa Media Player to the latest version.

Not sure how to get my 2FA key now as well, from Amazon right?  Followed all the steps and I cannot find it but 2FA is in place. sigh...

Appreciate any possible solutions.  I liked this integration!

**Reply by danielbrunt57** [reply]
> Not sure how to get my 2FA key now as well, from Amazon right? Followed all the steps and I cannot find it but 2FA is in place. sigh...

You can't get it after the fact from Amazon.  If your authenticator app does not yet you see the TOTP secret and you failed to save it when the QR code was first presented, you are SOL with that one.  However, you can just add another app to your Amazon account and capture the secret for its QR code when it comes up.

**Comment by malballuk1** [comment]
Mine is broken after todays HA update, however, my Alexa scripts still run even though Media Player is showing failed to set up

**Reply by Armand29** [reply]
> Mine is broken after todays HA update, however, my Alexa scripts still run even though Media Player is showing failed to set up

Mine did the same but a full reboot seemed to fix it.

**Reply by mario-pirozzi** [reply]
Same error with 2024.8. I modified the file  manifest .json --> "alexapy==1.28.2" + reboot, and it seems to work but it's not stable.
![Screenshot 2024-08-08 160121](https://github.com/user-attachments/assets/411224af-017a-45d0-8dd6-d34180f49ffb)

**Comment by malballuk1** [comment]
I have done a full reboot of my Pi but still getting the error but Alexa still seems to be working

**Comment by Boeng111** [comment]
Hello there,

after running into the same issues like described in the posts before, i completely removed the integration from my HA instance and re-installed it. It wouldn't correctly install yesterday but today it seemed to work. At least i got no errors, the integration showed up correctly and all of my devices were added to my HA again.
But sadly now all Devices show up as 'not available' and nothing i tried till now got them to work properly. Did i miss something?
Would be glad to support mith any needed information since i used this integration constantly to send around announcements and it would be damn sad we don't get it to work any more.

**Reply by Boeng111** [reply]
So i found the solution myself, maybe it helps someone. I deactiveted the integration and activated it again with 'amazon.de' instead of 'amazon.com'. (i live in Germany, of course)
It wouldn't install with '.de' yesterday so i tried '.com' today and after everything was installed fine and alle devices were added, i didn't thought of that issue at first. Changing it to '.de' installed the integration fine and added all devices as before, but now those are available and working correctly.
Thanks for the work on this integration, it's of great use for me.

**Reply by danielbrunt57** [reply]
> Did i miss something? 

Try deleting the pickle file in .storage and restarting HA.

**Comment by mario-pirozzi** [comment]
Sorry, but it after still not working after yesterday's update where "alexapy==1.28.2" was fixed. 
Now in log is:  **An error occurred accessing AlexaAPI: An exception of type AlexapyLoginError occurred. Arguments: ('Login error detected; not contacting API',)**
To make it work you have to go to reload devices every time.
does it happen to you too?
Thanks for the work on this integration.

**Comment by solarjunkev** [comment]
I've uploaded 4.12.7 (it now works) and sat on 2024.7.4 (I haven't upgraded HAOS to the August release yet)
The integration is working fine.
Kindly let me know when it's safe to load the new August HA firmware release as the AMP integration is a crucial part of my HA project.
Cheers

**Comment by labanex** [comment]
Thank you for fixing this. It was only a few days, not bad at all. Love this plugin.

One request: On the HACS page for this addon, I do not see a link to buy you coffee. Please add this feature as it is lacking a way for any of us to donate towards this plugin. This is a lacking feature and I will only give you five stars. I would give you 10 stars but it is lacking the donation button. At least, let me buy you a cup of coffee, okay?!!! :) 

I was trying to be funny. If there is a link, I do not see it. lol

**Comment by malballuk1** [comment]
Yes thanks, latest update seems to have fixed it for me too, nice one

**Comment by solarjunkev** [comment]
Is it safe to upgrade to 2024.8.x.yet? thanks

**Comment by nicedevil007** [comment]
so I have 2024.8.1 running, latest OS Update that was released today (version 13 if I'm right).
I reinstalled alexa media player from scratch, I remove the pickle file in .storage nothing helps :(

Always 403 forbidden

**Reply by danielbrunt57** [reply]
> I reinstalled alexa media player from scratch, I remove the pickle file in .storage nothing helps :(
> 
> Always 403 forbidden

@nicedevil007 I'm unable to comment without seeing your debug logs...

**Reply by nicedevil007** [reply]
> > I reinstalled alexa media player from scratch, I remove the pickle file in .storage nothing helps :(
> > Always 403 forbidden
> 
> @nicedevil007 I'm unable to comment without seeing your debug logs...

Sorry for that, I don't know where to get these logs if the integration isn't installed :( Can you help me with that?

**Reply by danielbrunt57** [reply]
> Sorry for that, I don't know where to get these logs if the integration isn't installed :( Can you help me with that?

They'll be in same place all other HA logs.  Just filter the full log display with `alexa`.
Make sure you have both `alexapy` and `custom_components/alexamedia` set to `debug`.
My configuration.yaml is split out into files:

![image](https://github.com/user-attachments/assets/5d9ad0f6-2e34-4b25-baba-15d75dfd5934)

If it's all in `configuration.yaml, it would look like...
```
logger:
  default: info
  logs:
    alexapy: debug
    custom_components.alexa_media: debug
```

![image](https://github.com/user-attachments/assets/bfba9b6b-df96-4370-96f4-225512e74de5)

![image](https://github.com/user-attachments/assets/d8d81104-47f3-4941-8a64-6292d6a27482)

![image](https://github.com/user-attachments/assets/c544a516-b4ad-4106-8a26-1ad46c51545f)

![image](https://github.com/user-attachments/assets/a2ce4247-6c28-4701-9d31-0e6b67bfbb09)

**Reply by nicedevil007** [reply]
here we go :)

I'm not entering the 2FA in the config during the integration installation, I do it (at that it what I would have done as many years before) in the upcoming browsertab where I have to enter everything.

**Reply by danielbrunt57** [reply]
> I'm not entering the 2FA in the config during the integration installation, I do it (at that it what I would have done as many years before) in the upcoming browsertab where I have to enter everything.

AMP has no idea what you are doing in the separate browser tabs!  Those are strictly between you (your browser) and Amazon. Your procedure will not work as AMP has no way to authenticate with Amazon EVER since Amazon does not permit legacy email/password only logins when Two Step Verification is enabled on the account.  You must provide AMP with the OTP secret so that it can generate the OTP code on its own.  That's the reason the AMP auth verification process requests confirmation that the OTP it's about to provide to Amazon is correct and that it can safely store the OTP secret and use it in future internal authentications with Amazon!

**Reply by nicedevil007** [reply]
so you mean I have to enter the seed code that generates the 2FA 6 digit codes, OK I have done this, now I get this message
![image](https://github.com/user-attachments/assets/f7d427ac-067a-44e5-8087-4fdd179b65ae)

I verify that the code is correct on my 2FA app and it is (and there are 15sec + left before it changes again).

Still 403 :( Log looks like this:

```bash
2024-08-16 14:26:04.564 DEBUG (MainThread) [custom_components.alexa_media.config_flow] Creating new login
2024-08-16 14:26:04.564 DEBUG (MainThread) [custom_components.alexa_media.helpers] a****n@g***********e: Returning uuid {'uuid': 'f94f7e499ee4048c8ee30633983e8469', 'index': 0}
2024-08-16 14:26:04.570 DEBUG (MainThread) [alexapy.alexalogin] Creating TOTP for Z************************************************6OA
2024-08-16 14:26:04.571 DEBUG (MainThread) [alexapy.alexalogin] Generating OTP 811666
2024-08-16 14:26:04.571 DEBUG (MainThread) [alexapy.alexalogin] Login created for a*******@*******de - amazon.de
2024-08-16 14:26:04.584 DEBUG (MainThread) [alexapy.alexalogin] Generating OTP 811666
2024-08-16 14:26:04.584 DEBUG (MainThread) [custom_components.alexa_media.config_flow] Generating OTP from 811666
2024-08-16 14:26:12.766 DEBUG (MainThread) [custom_components.alexa_media.config_flow] Starting proxy for a****n@g***********e - amazon.de
2024-08-16 14:26:12.766 DEBUG (MainThread) [alexapy.alexalogin] Attempting oauth login to https://www.amazon.com/ap/signin?openid.return_to=https://www.amazon.com/ap/maplanding&openid.assoc_handle=amzn_dp_project_dee_ios&openid.identity=http://specs.openid.net/auth/2.0/identifier_select&pageId=amzn_dp_project_dee_ios&accountStatusPolicy=P1&openid.claimed_id=http://specs.openid.net/auth/2.0/identifier_select&openid.mode=checkid_setup&openid.ns.oa2=http://www.amazon.com/ap/ext/oauth/2&openid.oa2.client_id=device:663934663765343939656534303438633865653330363333393833653834363932333431333234393536346335363335353634643332353733383331&openid.ns.pape=http://specs.openid.net/extensions/pape/1.0&openid.oa2.response_type=code&openid.ns=http://specs.openid.net/auth/2.0&openid.pape.max_auth_age=0&openid.oa2.scope=device_auth_access&openid.oa2.code_challenge_method=S256&openid.oa2.code_challenge=tiA867xJnaCpf70tNOU9iM_iGFsCxF9xK9EP6PDhecg&language=de_DE
2024-08-16 14:26:12.775 DEBUG (MainThread) [alexapy.alexalogin] Generating OTP 811666
2024-08-16 14:26:12.775 DEBUG (MainThread) [alexapy.alexalogin] Generating OTP 811666
```

**Reply by nicedevil007** [reply]
I tested it again with a complete removal of my 2FA stuff on amazon and recreating one new 2FA code. Still the same :(

EDIT: For sience: I removed the 2FA completly and then tried again without providing anything besides username/password and right localtion URL. same!

**Reply by nicedevil007** [reply]
ok I was just "stupid"... layer 8 problem ;)
call me a bonehead or whatever you want

this is how the integration looks like on the setup page before I enter anything...
![image](https://github.com/user-attachments/assets/de329973-4def-4dc5-bcb2-43ee97af22f8)


you see my local and WAN DNS here. I always change the upper one to the WAN one and the lower one to the local one. BUT there is no http**s** in the front of the URL. Long story short that was the issue for 403 :)

**Comment by mikemccomas** [comment]
Broken again. It seems like it breaks every week. Unfortunately, it messes up my other automations. I get it, it's free and people are trying their best to maintain it in their spare time as a passion project and unless I'm willing to start committing some code, I have zero room to complain. I'm just wondering if it's something that Amazon is doing with their API to make it difficult to keep this thing going?

**Comment by giancn** [comment]
Hi, any news?
I have installed HA Core v2024.8.3 and AMP v4.12.12 but I can't get it to work. I tried deleting and redownloading AMP but it won't let me authenticate anymore. It keeps asking me for Amazon user and password and it won't reconfigure
Anyone have the same problem?

**Reply by tripa3** [reply]
Have exactly the same problem, my HDD on my HA failed, and had to start again, now i get the same issue over and over, stuck on Welcome, sign in Already a customer? loop. if i anyone has any ideas that would be great.

**Reply by danielbrunt57** [reply]
> Have exactly the same problem, my HDD on my HA failed, and had to start again, now i get the same issue over and over, stuck on Welcome, sign in Already a customer? loop. if i anyone has any ideas that would be great.

See issue #2514, specifically [my workaround](https://github.com/alandtse/alexa_media_player/issues/2514#issuecomment-2345070532) which is to select Create a new account which will fail, then reselect login to existing account which will then give you a window with both email & password on the same screen like it used to and then the OTP window will follow after that.  Amazon has changed the login window structure and AMP is now befuddled with the email, then password, then OTP sequence as it expects email + password, then OTP, then return back to me.

**Reply by giancn** [reply]
Many thanks, I will try the workaround as soon as possible. In the meantime, I downgraded HA to version 2024.7.4 and everything works again. Regards

**Reply by tripa3** [reply]
thanks @danielbrunt57, it did the trick, i have added Alexia Media Player to HA with your work around. Thanks again for your help.

**Reply by ckrause7** [reply]
@danielbrunt57 Your workaround worked for me too after agonizing hours of trying. Many many thanks!

**Reply by morphodone** [reply]
@danielbrunt57 Thank you for the fix!

**Reply by KLiNG0NE** [reply]
With your instructions I was able to log in again. Many thanks for that! –
Now my Home Assistant restarts regularly without me doing anything.
(Home Assistant 2024.10.0 – Alexa Media Player 4.13.3)

Edited:
I was able to solve the problem by reverting to version 4.13.2 of Alexa Media Player. Version 4.13.3 definitely causes the restarts.

**Comment by zav13r** [comment]
The workaround seems valid for the login loop, but now I have a blank 'Error' dialogue back in HA at the final step. Is anyone else experiencing this?

**Comment by yettavr6** [comment]
Not really working for me. I only need it for 2 light switches that are Alexa-only (Eaton) but this point I’m just going to buy new HomeKit or Matter light switches because they’re not available through this integration. Only the Alexa speakers are available which isn’t really useful.
